astropy/io/misc/tests/test_hdf5.py

```diff
@@ -567,6 +567,41 @@ def test_read_h5py_objects(tmpdir):
     f.close()         # don't raise an error in 'test --open-files'
 
 
+@pytest.mark.skipif('not HAS_H5PY')
+def test_write_to_Hdf5(tmpdir):
+    test_file = str(tmpdir.join('test.hdf5'))
+    t = Table()
+    t['p'] = ['a', 'b', 'c']
+    t['q'] = [ 1, 2, 3 ]
+    t['r'] = [b'a', b'b', b'c']
+    t['s'] = ["\u2119", "\u01b4", "\u2602"]
+    t.write(test_file, path='the_table', overwrite=True)
+    t1 = Table.read(test_file, path='the_table', character_as_bytes=False)
```

Comment:
Can you re-order the test to make it easier to follow.  For each of `t`, `t1`, `t2`, `t3` have a distinct block that defines the table, then tests both the content and dtypes.